index.js

```diff
@@ -2,15 +2,26 @@
 
 const fp = require('fastify-plugin')
 const { Cache } = require('async-cache-dedupe')
+const S = require('fluent-json-schema')
+const Ajv = require('ajv')
+const ajv = new Ajv()
```

Comment:
Regarding the origin issue and this check:

> might be tricky to do but I would definitely look into it and if anybody has inputs about how to match two schemas to see that they "intersect" please share!

It can be done before the `setupSchema` execution.

Accessing the data through:

```js
graphQLSchema.getQueryType().getFields()
graphQLSchema.getMutationType().getFields()
```

It is possible to check if a cache `Query.fooBar: true` is missing from the schema.
In this way, you can validate that all the cachable queries have a resolver associated.